tfx/components/transform/executor_test.py

```diff
@@ -41,10 +43,39 @@ class _TempPath(types.Artifact):
 # TODO(b/122478841): Add more detailed tests.
 class ExecutorTest(tft_unit.TransformTestCase):
 
-  def _get_source_data_dir(self):
-    return os.path.join(
+  @classmethod
+  def setUpClass(cls):
+    cls._temp_data_dir = tempfile.mkdtemp()
+
+    source_data_dir = os.path.join(
         os.path.dirname(os.path.dirname(__file__)), 'testdata')
 
+    source_example_dir = os.path.join(source_data_dir, 'csv_example_gen')
+    source_schema_dir = os.path.join(source_data_dir, 'schema_gen')
+    source_module_dir = os.path.join(source_data_dir, 'module_file')
+
+    io_utils.copy_dir(source_schema_dir,
+                      os.path.join(cls._temp_data_dir, 'schema_gen'))
+    io_utils.copy_dir(source_module_dir,
+                      os.path.join(cls._temp_data_dir, 'module_file'))
```

Comment:
is that necessary to copy schema and model_file?

we just need to create a new input examplegen artifact2 that point to the copy location, other file path can remain the same?

Reply:
yup, fixed this. doing some additional function cleaning